### ISSUE TITLE
Refactor: ContextPage component is very long and could benefit from some refactoring

### ISSUE THREAD

**Issue by jayeclark** [mikhael28/paretOS#156]
## Description

The component `context/ContextPage.js` clocks in at a whopping 569 lines of code! Looking through, a good amount of it appears to be repetitive code that could be refactored.

## Goals
- [ ] Reduce the total lines of code 
- [ ] Reduce the amount of repetitive code
- [ ] Improve code efficiency and readability 
- [ ] Simplify state management for states that are always all updated at the same time
- [ ] Add comments as necessary to help orient new contributors navigating the file for the first time.

**Comment by syedinay** [comment]
Hello @jayeclark  I would like to pick this up

**Comment by jayeclark** [comment]
@syedinay Awesome, assigning it to you now! Let me know if you have any questions about the repo or the issue.

**Comment by syedinay** [comment]
@jayeclark  I have set up the project in my local. Do we have common credentials that I can use to sign in?

**Comment by mikhael28** [comment]
@syedinay you just make an account!

**Comment by syedinay** [comment]
@mikhael28  While signing up, I noticed that we have dummy terms and conditions pop-up which is totally fine, but the font color is light gray and the background is white, maybe we can create an issue for this and fix it?

**Comment by jayeclark** [comment]
Hey @syedinay, great eye! We switched to a dark mode UI a few weeks back and there is definitely still some cleanup needed around the site.

As luck would have it, I was working on the sign up form and noticed the same thing - I actually have a PR open that fixes it, #240  (just waiting on a second pair of eyes). 😁

**Comment by frankykubo** [comment]
Hello, is this issue still relevant? I would like to contribute if its available as a part of my study examination. Thank you!

**Comment by syedinay** [comment]
Hello @frankykubo, you can pick it up if you like. I am unable to find time to work on this.

**Comment by frankykubo** [comment]
Okay. Please assign this to me and I'll start working on this. May I fork from master, or some other branch?

**Comment by mikhael28** [comment]
@frankykubo you fork the repository, then in your fork - branch off from main. Were you able to get the repository up and running on your local machine?

**Comment by frankykubo** [comment]
> @frankykubo you fork the repository, then in your fork - branch off from main. Were you able to get the repository up and running on your local machine?

@mikhael28 I was able to get the repo working locally! I'll start soon and will be giving information continuously right here. Thanks!

**Comment by frankykubo** [comment]
Hello, may I ask, if the tests for this project are working and if there is a chance I can write any tests for the ContextPage component? And also, how to I actually get to /context/:id route through web ui? Thank you!

**Comment by frankykubo** [comment]
Hello, may I ask, if the tests for this project are working and if there is a chance I can write any tests for the ContextPage component? Thank you!

**Comment by jeremydthomas** [comment]
@mikhael28

**Comment by mikhael28** [comment]
Err the tests aren't really working at the moment, due to a migration to Vite and vitest from Jest. If you would like to add some tests, then that's great - you should be using the LTS version of Node, version 16.

**Comment by mikhael28** [comment]
If you would like to try and fix the tests, that would be swell too.

**Comment by frankykubo** [comment]
@mikhael28 Would it be okay if I separated ContextPage.tsx into ContextPageHubs.tsx and ContextPageGeneric.tsx ? Or you dont want to separate?

**Comment by frankykubo** [comment]
For now, I decided not to split ContextPage into two separate components. Here is what I did:

### Little design updates:
The header of Context page, especially the elements "Suggest new resource" and "Read overview" looked like plain text. I decided to enliven it a little. Suggest new resource button by adding outline and subtle background on hover, Read overview by adding pointer cursor for better UX and also opacity on hover.

#### Before after comparison:
**Before:**
![original](https://user-images.githubusercontent.com/50501921/199763764-fa1ec6d6-40dd-4353-9252-8927dbf82ff3.png)

**After (no hover)**
![normal](https://user-images.githubusercontent.com/50501921/199763949-e921719d-c29a-4bf0-9ee5-0f127833e8aa.png)

**After (hover)**
![hover](https://user-images.githubusercontent.com/50501921/199764062-f55ae3c0-8fd1-46cf-8661-b208b1cea697.png)


Because of hover styles, I made a new file @/css/buttons.css and imported it in index.jsx.

### Code refactor:
In this refactor, I was trying, where possible, to use ES6 and newer syntax, with arrow functions, single-line conditions...
I was able to lessen component's lines of code by ~100. So before, it was 570, after refactor, it is 472.

### Tests:
I dedicated one whole day to make the test work (Vitest, Jest, React-testing-library). I really wanted to make component testing for particular component, but I was not able to provide all props, data, etc. to make test running. At least, I decided to use Cypress E2E, which tests ContextPage component by navigating to Library of context's elements and testing content. The test is not part of the pull request, because it is out of scope of this issue.
